examples/CMakeLists.txt

```diff
@@ -1,25 +1,64 @@
 option(GINKGO_BUILD_EXTLIB_EXAMPLE "Build the external-lib-interfacing with deal.II, you need to link the deal.II library." OFF)
-add_subdirectory(custom-logger)
-add_subdirectory(custom-matrix-format)
-add_subdirectory(custom-stopping-criterion)
+option(GINKGO_RUN_EXAMPLES "Run the examples." OFF)
+
+function(setup_example_target target_name)
+    add_executable(${target_name} ${target_name}.cpp)
+    target_link_libraries(${target_name} ginkgo)
+    target_include_directories(${target_name} PRIVATE ${PROJECT_SOURCE_DIR})
+endfunction()
+
+set(EXAMPLES_EXEC_LIST "adaptiveprecision-blockjacobi;custom-logger;ginkgo-overhead;ginkgo-ranges;ilu-preconditioned-solver;ir-ilu-preconditioned-solver;inverse-iteration;iterative-refinement;mixed-precision-ir;nine-pt-stencil-solver;poisson-solver;preconditioned-solver;simple-solver;three-pt-stencil-solver;")
+
+set(EXAMPLES_LIST "${EXAMPLES_EXEC_LIST};custom-matrix-format;custom-stopping-criterion;minimal-cuda-solver;papi-logging;performance-debugging;preconditioner-export;simple-solver-logging")
+
 if(GINKGO_BUILD_EXTLIB_EXAMPLE)
-    add_subdirectory(external-lib-interfacing)
+    set(EXAMPLES_LIST "${EXAMPLES_LIST};external-lib-interfacing")
 endif()
-add_subdirectory(adaptiveprecision-blockjacobi)
-add_subdirectory(ginkgo-overhead)
-add_subdirectory(ginkgo-ranges)
-add_subdirectory(ilu-preconditioned-solver)
-add_subdirectory(ir-ilu-preconditioned-solver)
-add_subdirectory(inverse-iteration)
-add_subdirectory(iterative-refinement)
-add_subdirectory(minimal-cuda-solver)
-add_subdirectory(mixed-precision-ir)
-add_subdirectory(nine-pt-stencil-solver)
-add_subdirectory(papi-logging)
-add_subdirectory(performance-debugging)
-add_subdirectory(poisson-solver)
-add_subdirectory(preconditioned-solver)
-add_subdirectory(preconditioner-export)
-add_subdirectory(simple-solver)
-add_subdirectory(simple-solver-logging)
-add_subdirectory(three-pt-stencil-solver)
+
+foreach(example ${EXAMPLES_LIST})
+    add_subdirectory(${example})
+endforeach()
+
+set(run_list "")
+if(GINKGO_RUN_EXAMPLES)
+    foreach(example ${EXAMPLES_LIST})
+        add_custom_target("run-${example}"
+            COMMAND ${CMAKE_CURRENT_BINARY_DIR}/${example}/${example} > ${CMAKE_CURRENT_BINARY_DIR}/${example}/${example}.out
+            WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/${example})
+    endforeach()
+    foreach(example ${EXAMPLES_LIST})
+        file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/${example}/diff-command "#!/bin/bash
+diff <(sed -n '7,$p' ${CMAKE_CURRENT_BINARY_DIR}/${example}/${example}.out | sed -E 's/([0-9]+.)//g') <(sed -n '6,$p' ${CMAKE_SOURCE_DIR}/examples/${example}/doc/results.dox | head -n -4 | sed -E 's/([0-9]+.)//g')")
```

Comment:
Instead of having a global diff command, we could also add individual scripts to the examples that do nothing else but read from stdin and compare the output to what we expect. For most examples, this could be a simple residual check, and for the more complex outputs, we could do some kind of pattern matching with expected output patterns.

Reply:
I do not agree here. In my view, the examples shouldn't really be integration tests. What we want with this system is to only make sure that examples still run successfully after every Ginkgo update which I believe this sort of check is enough.

Reply:
If I recall correctly, this effort was sparked at least in part by the fact that that the output of some examples was many orders off from what the documentation says during the JOSS review, so this would be a really cheap way to avoid such issues in the future.
I do agree that this is far (both in purpose and execution) from integration tests.

Reply:
The main problem there was with the twenty-seven-pt stencil solver, which was actually wrong. I think the integration tests should definitely more comprehensive with testing the residuals and the convergence. But maybe for examples, i think this is sufficient.